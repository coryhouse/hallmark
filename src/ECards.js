import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

class ECards extends React.Component {
  renderTable() {
    return (
      <table>
        <thead>
          <tr>
            <th>&nbsp;</th>
            <th>SKU</th>
            <th>Name</th>
            <th>Category</th>
          </tr>
        </thead>
        <tbody>
          {this.props.eCards.map(card => {
            return (
              <tr key={card.sku}>
                <td>
                  <button
                    onClick={event => this.props.onDeleteClick(event, card.sku)}
                  >
                    Delete
                  </button>
                </td>
                <td>{card.sku}</td>
                <td>{card.name}</td>
                <td>{card.category}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
    );
  }

  render() {
    return (
      <React.Fragment>
        <Link to="/manage">Add eCard</Link>
        {this.props.eCards.length === 0 ? (
          <div>
            No Cards{" "}
            <span role="img" aria-label="meh">
              ðŸ˜œ
            </span>
          </div>
        ) : (
          this.renderTable()
        )}
      </React.Fragment>
    );
  }
}

ECards.propTypes = {
  eCards: PropTypes.array.isRequired,
  onDeleteClick: PropTypes.func.isRequired
};

export default ECards;
