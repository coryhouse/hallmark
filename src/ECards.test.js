import ECards from "./ECards";
import React from "react";
import { create } from "react-test-renderer";

describe("ECards", () => {
  it("should show no cards message when no cards exist", () => {
    // Act
    const tree = create(
      <ECards eCards={[]} onAddClick={jest.fn()} onDeleteClick={jest.fn()} />
    ).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it("should show cards table when cards exist", () => {
    // Act
    const tree = create(
      <ECards
        eCards={[{ sku: 1, name: "Cory, You're Amazing", category: "Truth." }]}
        onAddClick={jest.fn()}
        onDeleteClick={jest.fn()}
      />
    ).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
