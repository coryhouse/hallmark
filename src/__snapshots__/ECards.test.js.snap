// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECards should show cards table when cards exist 1`] = `
Array [
  <button
    onClick={[Function]}
>
    Add eCard
</button>,
  <table>
    <thead>
        <tr>
            <th>
                Â 
            </th>
            <th>
                SKU
            </th>
            <th>
                Name
            </th>
            <th>
                Category
            </th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                <button
                    onClick={[Function]}
                >
                    Delete
                </button>
            </td>
            <td>
                1
            </td>
            <td>
                Cory, You're Amazing
            </td>
            <td>
                Truth.
            </td>
        </tr>
    </tbody>
</table>,
]
`;

exports[`ECards should show no cards message when no cards exist 1`] = `
Array [
  <button
    onClick={[Function]}
>
    Add eCard
</button>,
  <div>
    No Cards
     
    <span
        aria-label="meh"
        role="img"
    >
        ðŸ˜œ
    </span>
</div>,
]
`;
